### PR TITLE
(DOCSP-22347): Flutter additional auth provider

### PR DESCRIPTION
**Pull request by mongodben**
## Pull Request Info

Documentation on following additional auth providers:

- multiple anonymous
- custom JWT
- custom function
- apple
- google (2 ways)
- facebook

### Jira

- https://jira.mongodb.org/browse/DOCSP-22347

### Staged Changes (Requires MongoDB Corp SSO)

- [Authenticate (Flutter)](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-22347/sdk/flutter/users/authenticate/)

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: source/sdk/flutter/users/authenticate.txt [hunk 1]
```diff
@@ -62,6 +62,12 @@ and then pass the generated credential to
 .. literalinclude:: /examples/generated/flutter/authenticate_users_test.snippet.anonymous-credentials.dart
    :language: dart
 
+If you want more than one anonymous users, set ``reuseCredentials: false``
```
Comment (dacharyc) [inline source/sdk/flutter/users/authenticate.txt:65]:
```suggestion
If you want more than one anonymous user, set ``reuseCredentials: false``
```

File: examples/dart/test/authenticate_users_test.dart [hunk 2]
```diff
@@ -47,6 +60,78 @@ void main() {
...
+      User currentUser = await app.logIn(jwtCredentials);
+      // :snippet-end:
+      expect(currentUser.provider, AuthProviderType.jwt);
+      // clean up
+      app.deleteUser(currentUser);
+    });
+    test("Custom Function user", () async {
+      // :snippet-start: custom-function-credentials
+      Map<String, String> credentials = {
+        "username": "iloverealm",
```
Comment (dacharyc) [inline examples/dart/test/authenticate_users_test.dart:95]: Just a nit; I think we should avoid things like an "iloverealm" username, given the issues we had with a similar MongoDB string. I don't think anyone would necessarily find this offensive, but it's probably better to just use a name or something like "someUsername".

**Review by dacharyc — APPROVED** [review]
Couple of tiny nits, but overall LGTM.

File: examples/dart/test/authenticate_users_test.dart (hunk 2 above)
Comment (mongodben) [reply examples/dart/test/authenticate_users_test.dart:95]:
```suggestion
        "username": "someUsername",
```

File: source/examples/generated/flutter/authenticate_users_test.snippet.custom-function-credentials.dart [hunk 3]
```diff
@@ -0,0 +1,8 @@
+Map<String, String> credentials = {
+  "username": "iloverealm",
```
Comment (mongodben) [inline source/examples/generated/flutter/authenticate_users_test.snippet.custom-function-credentials.dart:2]:
```suggestion
  "username": "someUser",
```

**Pull request merged**